core/block_animations.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2018 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Methods animating a block on connection and disconnection.
+ * @author fenichel@google.com (Rachel Fenichel)
+ */
+'use strict';
+
+goog.provide('Blockly.BlockAnimations');
+
+
+/**
+ * PID of disconnect UI animation.  There can only be one at a time.
+ * @type {number}
+ * @private
+ */
+Blockly.BlockAnimations.disconnectPid_ = 0;
+
+/**
+ * SVG group of wobbling block.  There can only be one at a time.
+ * @type {Element}
+ * @private
+ */
+Blockly.BlockAnimations.disconnectGroup_ = null;
+
+/**
+ * Play some UI effects (sound, animation) when disposing of a block.
+ * @param {!Blockly.BlockSvg} block The block being disposed of.
+ * @package
+ */
+Blockly.BlockAnimations.disposeUiEffect = function(block) {
+  var workspace = block.workspace;
+  var svgGroup = block.getSvgRoot();
+  workspace.getAudioManager().play('delete');
+
+  var xy = workspace.getSvgXY(svgGroup);
```

Comment:
how come you can drop the type annotation? I guess I don't quite know why it was needed before.

Reply:
I was a cast before (from the perspective of closure-compiler).  `getSvgRoot` is already annotated to return the correct type.  

It might eventually need a cast inside it, but that's out of scope for right now.  It gives warnings, not errors.